core/src/test/java/org/apache/accumulo/core/file/rfile/GenerateSplitsTest.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.file.rfile;
+
+import static org.apache.accumulo.core.file.rfile.GenerateSplits.main;
+import static org.apache.accumulo.core.file.rfile.RFileTest.newColFamByteSequence;
+import static org.apache.accumulo.core.file.rfile.RFileTest.newKey;
+import static org.apache.accumulo.core.file.rfile.RFileTest.newValue;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.List;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+@SuppressFBWarnings(value = "PATH_TRAVERSAL_IN", justification = "paths provided by test")
+public class GenerateSplitsTest {
+  private static final Logger log = LoggerFactory.getLogger(GenerateSplitsTest.class);
+
+  @ClassRule
```

Comment:
This is the first ever occurrence of `@ClassRule` in our code. We usually just use `@Rule`. I'm guessing you did this so you only need to create test RFiles once, to be shared by all tests. I'm wondering if there's any downside to having each test reuse the same temp folder and test file.

Reply:
I can't think of one for this test since I am only reading the file.